packages/yourloops/app/error.tsx

```diff
@@ -48,17 +52,36 @@ interface OnErrorProps {
   error?: Error
 }
 
+const classes = makeStyles(() => ({
+  errorId: {
+    display: 'block',
+    fontWeight: 'bold'
+  }
+}))
+
 function OnError(props: OnErrorProps): JSX.Element {
   const { t } = useTranslation('yourloops')
   const theme = useTheme()
+  const location = useLocation()
   const [showMore, setShowMore] = React.useState(false)
   const fullScreen = useMediaQuery(theme.breakpoints.down('md'))
-
+  const errorId = uuidv4()
+  const style = classes()
   const errorMessage = props.error?.message ?? 'n/a'
+  const error = props.error ? `Error: ${errorMessage}\nStack: ${props.error.stack}` : 'N/A'
+  const info = `${(props.event as string).toString()}\nSource: ${props.source}:${props.lineno}:${props.colno}\n${error}`
 
   React.useEffect(() => {
     try {
       metrics.send('error', 'app-crash', errorMessage)
+      ErrorApi.sendError({
+        browserName,
+        browserVersion,
+        date: moment(new Date()).format('DD/MM/YYYY'),
+        err: errorMessage,
+        errorId,
+        path: location.pathname
+      }).catch(error => console.error(error))
```

Comment:
I don't understand why there is a need for both `ErrorApi.sendError().catch()` and `try {} catch {}`? Besides, they seem to be doing the same thing

Reply:
The try catch cannot catch the error coming from `ErrorApi.sendError(...)`.
And the try catch is used to get the error for the `metrics.send(...)`
But I think that I can improve this as our http service seems to log the error too. :slightly_smiling_face: